rtos/Mutex.h

```diff
@@ -28,11 +28,26 @@
 #include "mbed_rtos_storage.h"
 
 #include "platform/NonCopyable.h"
+#include "platform/ScopedLock.h"
 
 namespace rtos {
 /** \addtogroup rtos */
 /** @{*/
 
+class Mutex;
+/** Typedef for the mutex lock
+ *
+ * Usage:
+ * @code
+ * void foo(Mutex &m) {
+ *     ScopedMutexLock lock(m);
+ *     // Code in this block will be protected by Mutex lock
+ * }
+ * @endcode
+ */
+typedef mbed::ScopedLock<Mutex> ScopedMutexLock;
```

Comment:
My only concern for this one is that `Mutex::lock()` returns CMSIS-RTOS errors, which `ScopedLock` will ignore. I'd feel happier if `lock` was a `void` that asserted or mbed_errored on CMSIS-RTOS errors.

But I guess we're largely okay because most of those errors will be caught by our `EvrRtxMutexError` handler in a debug build - a lot of our RTOS functions are now effectively limited in what errors they can return, despite their documentation.

Reply:
I can add a partial specialization of `ScopedLock` that asserts on `lock` and `unlock` if they return an `osStatus`. Would that work for you ?

Reply:
That would work. I'd prefer a variant of `Mutex` with revised API, or revising `Mutex` itself, but a specialisation of `ScopedLock` addresses it easily here for now, and means any transformations from manual locking to `ScopedLock` will get this improvement.